x-pack/plugins/maps/public/layers/sources/mvt_single_layer_vector_source/mvt_single_layer_vector_source.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import uuid from 'uuid/v4';
+import { AbstractSource, ImmutableSourceProperty } from '../source';
+import { TiledVectorLayer } from '../../tiled_vector_layer';
+import { GeoJsonWithMeta, ITiledSingleLayerVectorSource } from '../vector_source';
+import { MAX_ZOOM, MIN_ZOOM, SOURCE_TYPES } from '../../../../common/constants';
+import { VECTOR_SHAPE_TYPES } from '../vector_feature_types';
+import { IField } from '../../fields/field';
+import { registerSource } from '../source_registry';
+import { getDataSourceLabel, getUrlLabel } from '../../../../common/i18n_getters';
+import {
+  LayerDescriptor,
+  MapExtent,
+  TiledSingleLayerVectorSourceDescriptor,
+  VectorSourceRequestMeta,
+  VectorSourceSyncMeta,
+} from '../../../../common/descriptor_types';
+import { VectorLayerArguments } from '../../vector_layer';
+
+export const sourceTitle = i18n.translate(
+  'xpack.maps.source.MVTSingleLayerVectorSource.sourceTitle',
+  {
+    defaultMessage: 'Vector Tile Layer',
+  }
+);
+
+export class MVTSingleLayerVectorSource extends AbstractSource
+  implements ITiledSingleLayerVectorSource {
+  static createDescriptor({
+    urlTemplate,
+    layerName,
+    minSourceZoom,
+    maxSourceZoom,
+  }: TiledSingleLayerVectorSourceDescriptor) {
+    return {
+      type: SOURCE_TYPES.MVT_SINGLE_LAYER,
+      id: uuid(),
+      urlTemplate,
+      layerName,
+      minSourceZoom: Math.max(MIN_ZOOM, minSourceZoom),
+      maxSourceZoom: Math.min(MAX_ZOOM, maxSourceZoom),
+    };
+  }
+
+  readonly _descriptor: TiledSingleLayerVectorSourceDescriptor;
+
+  constructor(
+    sourceDescriptor: TiledSingleLayerVectorSourceDescriptor,
+    inspectorAdapters?: object
+  ) {
+    super(sourceDescriptor, inspectorAdapters);
+    this._descriptor = sourceDescriptor;
+  }
+
+  renderSourceSettingsEditor() {
+    return null;
+  }
+
+  getFieldNames(): string[] {
+    return [];
+  }
+
+  createDefaultLayer(options: LayerDescriptor): TiledVectorLayer {
+    const layerDescriptor = {
+      sourceDescriptor: this._descriptor,
+      ...options,
+    };
+    const normalizedLayerDescriptor = TiledVectorLayer.createDescriptor(layerDescriptor, []);
+    const vectorLayerArguments: VectorLayerArguments = {
+      layerDescriptor: normalizedLayerDescriptor,
+      source: this,
+    };
+    return new TiledVectorLayer(vectorLayerArguments);
+  }
+
+  getGeoJsonWithMeta(
```

Comment:
Can this be deleted? This no longer implements VectorSource so no need for the method